### ISSUE TITLE
onClose not called when closing by clicking outside of the picker

### ISSUE THREAD

**Issue by josdejong** [Sphinxxxx/vanilla-picker#5]
@Sphinxxxx thanks for writing and sharing this color picker, I like the clean and powerful API. I'm using it in [JSONEditor](https://github.com/josdejong/jsoneditor/) now and it works like a charm :)

I noticed that the `onClose` event is only called when you click "Ok" or click on the `parent` element where the editor is attached. I would expect `onClose` to be fired too when clicking anywhere outside of the editor but this is not the case. It would be great if there is a way to get triggered when the editor is closed either via ok or a cancel. Any ideas?

**Comment by Sphinxxxx** [comment]
Thanks a lot!

`onClose` should indeed be called whenever the popup closes, no matter how. Go to [this demo page](https://rawgit.com/Sphinxxxx/vanilla-picker/master/demo/index.html) and watch the console output when you toggle the "Click me!" picker.

If you still think there's an error, do you have a JSFiddle or similar to demonstrate it?

**Comment by josdejong** [comment]
Ah, you're right, `vanilla-picker` works correctly but it was something in my code blocking it: I temporarily create a color picker and clean it up from the DOM when closed (click outside), That cleanup was a bit too rigid, also blocking the event that `vanilla-picker` needs to close itself. My bad!

**Comment by Sphinxxxx** [comment]
Btw, if I understand your use case correctly:

Instead of creating and deleting a bunch of pickers, would it be more convenient to create one picker which you share (move) between all your color nodes? That was always a half-implemented feature, but is now released in v2.4.0 with the `movePopup()` method. Example:

```javascript

var pickerShared = new Picker({
    onChange: function(color) {
        this.settings.parent.style.backgroundColor = color.rgbaString;
    }
});

...

function onParentClick(e) {
    var newParent = e.target;

    pickerShared.movePopup({
        parent: newParent,
        color: newParent.style.backgroundColor,
    }, true);
});

```

(See the [demo page](https://rawgit.com/Sphinxxxx/vanilla-picker/master/demo/index.html) under the "Shared picker" heading).

**Comment by josdejong** [comment]
Thanks, that's indeed the use case, thanks for your suggestion. 

The JSONEditor contains a JSON document which can contain many color pickers, so the color picker is created lazily. Creating a single color picker and reusing it could be a nice optimization, though it's not really needed right now since creating a color picker takes hardly any time anyone.
